fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -582,14 +580,19 @@ private Catalog(boolean isCheckpointCatalog) {
         this.tabletScheduler = new TabletScheduler(this, systemInfo, tabletInvertedIndex, stat, Config.tablet_rebalancer_type);
         this.tabletChecker = new TabletChecker(this, systemInfo, tabletScheduler, stat);
 
+        // The pendingLoadTaskScheduler's queue size should not less than Config.desired_max_waiting_jobs.
+        // So that we can guarantee that all submitted load jobs can be scheduled without being starved.
         this.pendingLoadTaskScheduler = new MasterTaskExecutor("pending_load_task_scheduler", Config.async_pending_load_task_pool_size,
-                Config.async_pending_load_task_pool_size, !isCheckpointCatalog);
+                Config.desired_max_waiting_jobs, !isCheckpointCatalog);
+        // The loadingLoadTaskScheduler's queue size is unlimited, so that it can receive all loading tasks
+        // created after pending tasks finish. And don't worry about the high concurrency, because the
```

Comment:
pending task cost less time than loading task. so my idea is that if load task thread pool is full, not submit task of not started job, for that once start job and still no resource to execute loading task, the job will wait to be timeout, not because of slow running but absent of thread resource.

Reply:
OK. I changed. Now when submit a broker load job to the thread pool, I will check if there are available worker thread in `loading_load_task_pool`